scripts/bb-test-cleanup.sh

```diff
@@ -87,6 +97,17 @@ function upload_codecov_report_with_flag
         -C "$ZFS_REVISION" \
         -F "$1" \
         $PR_OR_BRANCH_OPT
+
+    #
+    # The compressed tarball will be collected and stored on the
+    # Buildbot "master". To try and reduce the amount of storage
+    # required to store these coverage reports, we do our best to
+    # compress the tarball as much possible. We measured xz to
+    # result in a file that's roughly half the size when compared
+    # to the output when using gzip; no other algorithms were tested.
+    #
+    tar -cf - "$CODE_COVERAGE_OUTPUT_DIRECTORY" | \
+        xz -9e "${WORKDIR}/${CODE_COVERAGE_OUTPUT_DIRECTORY}.tar.xz"
```

Comment:
Let's make this run conditionally on a `TEST_CODE_COVERAGE_HTML` variable.  That way we can leave it disabled by default while we work on the issues with PRs.

Reply:
Does `xz` require the full path as well considering it's being created in the current working directory?